src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java

```diff
@@ -238,6 +241,9 @@ public UpdateTranscriptionAdminResponse updateTranscriptionAdmin(Integer transcr
             transcriptionStatusEntity,
             updateTranscription.getWorkflowComment()
         );
+
+        auditApi.recordAll(auditActivityProvider);
```

Comment:
just for my knowledge, what is the benefit of having an `AuditApi` in addition to an `AuditService`? The service itself is providing an api for auditing.

Reply:
@mario-paniccia this was a pattern adopted fairly early on in the project. Exposing methods that are used outside of the package via the "Api" class (questionable naming). The idea behind this was to make it easier if we ever wanted to move packages out to some separate service, it would be easier to find and fix usages if they all go through one interface.

Reply:
I'm not sure really, I think was an attempt to segregate the domain areas in a DDD style but never really took off.  Again happy to look at this.

Reply:
I see. Seems a bit of boilerplate to me, and increases code complexity. But ok thanks for explaining this

Reply:
This was the original presentation, by @chrisbellingham-hmcts Not sure if it helps at all.

[access_patterns.pdf](https://github.com/user-attachments/files/15772555/access_patterns.3.pdf)